packages/dds/tree/src/test/feature-libraries/modular-schema/comparison.spec.ts

```diff
@@ -200,6 +210,169 @@ describe("Schema Comparison", () => {
 		);
 	});
 
+	// This helps provide some coverage for our schema evolution story, since repo compatibility
+	// influences the types of schema changes we allow
+	describe("allowsRepoSuperset", () => {
+		const compareTwoRepo = (a: TreeStoredSchema, b: TreeStoredSchema): boolean => {
+			return allowsRepoSuperset(defaultSchemaPolicy, a, b);
+		};
+
+		const createTestTree = (fields: [string, TreeFieldStoredSchema][]) => ({
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: new ObjectNodeStoredSchema(
+				new Map(fields.map(([key, schema]) => [brand(key), schema])),
+			),
+		});
+
+		it("Fix the rootFieldSchema and validate repo superset with different TreeNodeStoredSchemas", () => {
+			const rootFieldSchema = fieldSchema(FieldKinds.forbidden);
+
+			const testTrees = [
+				createTestTree([["x", fieldSchema(FieldKinds.required, [emptyTree.name])]]),
+				createTestTree([]),
+				createTestTree([["x", fieldSchema(FieldKinds.optional, [emptyTree.name])]]),
+				createTestTree([
+					["x", fieldSchema(FieldKinds.optional, [emptyTree.name])],
+					["y", fieldSchema(FieldKinds.optional, [emptyTree.name])],
+				]),
+				{ name: brand<TreeNodeSchemaIdentifier>("testTree"), schema: anyTreeWithoutValue },
+			];
+
+			const repos = testTrees.map((testTree) => {
+				const repo = new TreeStoredSchemaRepository();
+				repo.apply({
+					rootFieldSchema,
+					nodeSchema: new Map([[testTree.name, testTree.schema]]),
+				});
+				return repo;
+			});
+
+			testOrder(compareTwoRepo, repos);
+		});
+
+		it("Fix the TreeNodeStoredSchema and validate repo superset with different rootFieldSchemas", () => {
+			const rootFieldSchemas = [
+				fieldSchema(FieldKinds.required),
+				fieldSchema(FieldKinds.optional),
+				fieldSchema(FieldKinds.sequence),
+			];
+
+			const testTree = {
+				name: brand<TreeNodeSchemaIdentifier>("testTree"),
+				schema: valueAnyTree,
+			};
+
+			const repos = rootFieldSchemas.map((rootFieldSchema) => {
+				const repo = new TreeStoredSchemaRepository();
+				repo.apply({
+					rootFieldSchema,
+					nodeSchema: new Map([[testTree.name, testTree.schema]]),
+				});
+				return repo;
+			});
+
+			testOrder(compareTwoRepo, repos);
+		});
+
+		it("Validate the ordering when the identifiers are different", () => {
+			// TODO: There seems to be a bug in allowsRepoSuperset or allowsFieldSuperset,
```

Comment:
Can you track this with an item and follow up on it immediately? The bug may actually be in allowsTreeSuperset, as when the allowed types at the root aren't defined, `TreeTypeSet` is meant to encompass 'every allowed type in the schema', so we should probably have a check that 'every type allowed in argument 1 is a subset of those allowed in argument 2'. Seee comments on `TreeTypeSet` and `TreeFieldStoredSchema`.

Reply:
I created a work item [AB#8357](https://dev.azure.com/fluidframework/internal/_workitems/edit/8357) to follow-up within this sprint, perhaps we can create a separate PR to handle it